index/language-settings.go

```diff
@@ -0,0 +1,200 @@
+package index
+
+import (
+	"strings"
+
+	"golang.org/x/text/language"
```

Comment:
I agree with your comment re. the language package. I looked at the code, and it's written for the more complex use-case of language negotiation. We have a straight-forward matching case that could be solved with something like this: https://go.dev/play/p/q9e9qx1Nmpj